homeassistant/components/adguard/__init__.py

```diff
@@ -116,12 +122,14 @@ async def refresh(call: ServiceCall) -> None:
     return True
 
 
-async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_unload_entry(
+    hass: HomeAssistant, entry: ConfigEntry[AdGuardData]
+) -> bool:
     """Unload AdGuard Home config entry."""
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
     if unload_ok:
-        hass.data[DOMAIN].pop(entry.entry_id)
-    if not hass.data[DOMAIN]:
+        hass.data[DOMAIN] -= 1
+    if hass.data[DOMAIN] == 0:
```

Comment:
We don't need to store anything in hass.data here. We can just check if there's no other loaded config entry for the domain.

In the future this should all be removed since we no longer want to remove services when unloading config entries.